### ISSUE TITLE
azure-ai-documentintelligence samples for uploading base64

### ISSUE THREAD

**Issue by dkuska** [Azure/azure-sdk-for-python#33901]
**Is your feature request related to a problem? Please describe.**

I want to analyze base64 encoded documents through the new azure-ai-documentintelligence package.
It seems like I should do something like this:
```python

from azure.core.credentials import AzureKeyCredential
from azure.ai.documentintelligence import DocumentIntelligenceClient
from azure.ai.documentintelligence.models import AnalyzeResult, AnalyzeDocumentRequest

client = DocumentIntelligenceClient("my_endpoint", AzureKeyCredential("my_key"))

my_base64 = b"..."
poller = client.begin_analyze_document(
    "prebuilt-layout",
    analyze_request=AnalyzeDocumentRequest(base64_source=my_base64),
    content_type="application/octet-stream",
)
```

But no matter what I do I'm greeted with the following error: 
```
{
    "code": "InvalidContent",
    "message": "The file is corrupted or format is unsupported. Refer to documentation for the list of supported formats."
}
```
since the documentation for the `AnalyzeDocumentRequest` [Link](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py#L126) says that `base64_source` is of type `Optional[bytes]` I thought it'd work like this.

However all of the samples use this method either through:

1. Passing a _io.BufferedReader like so:
``` python
with open("path/to/local/file", "rb") as f:
    poller = client.begin_analyze_document(
        "prebuilt-layout",
        analyze_request=f,
        content_type="application/octet-stream",
    )
```
2. By passing a URL like so:
``` python
    url = "https://some.url/Invoice_1.pdf"
    poller = client.begin_analyze_document(
        "prebuilt-layout",
        AnalyzeDocumentRequest(url_source=url),
    )
```

**Describe the solution you'd like**

I'd like to get examples/tutorials for all the different options to upload to the document intelligence through the newly released package.
Especially interesting to me is uploading a base64 bytes object. 

**Describe alternatives you've considered**

I tried to use `from io import BytesIO` and create a buffer like this and use the `with BytesIO(my_base64) as buf` but this didn't work either

**Comment by kashifkhan** [comment]
Thank you for the feedback @dkuska . We will investigate and get back to you asap.

**Comment by YalinLi0312** [comment]
Hi @dkuska, I can reproduce the error, will investigate the root cause.

**Comment by YalinLi0312** [comment]
Update: I got the same error when I pass `mybase64=b"foo"`, it is not a bytes from a valid document. This is a doc you can refer: https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-analyze-document-response?view=doc-intel-4.0.0#analyze-document-request

**Comment by dkuska** [comment]
Hey, thanks for investigating this!

I did something like this to get the base64 bytes object:

```python 
with open(filepath, 'rb') as file:
    file_content = file.read()

mybytes64 =  bytes(base64.b64encode(file_content))

poller = client.begin_analyze_document(
    "prebuilt-layout",
    analyze_request=AnalyzeDocumentRequest(base64_source=mybytes64),
    content_type="application/octet-stream",
) 
```
Is this wrong? I wasn't for the life of me able to figure out what I did wrong...

**Comment by YalinLi0312** [comment]
Can you try to pass `file_content` without encode and not passing `content_type`? It would be:
```python
with open(filepath, 'rb') as file:
    file_content = file.read()

poller = client.begin_analyze_document(
    "prebuilt-layout",
    analyze_request=AnalyzeDocumentRequest(base64_source=file_content),
) 
```

**Comment by dkuska** [comment]
Hey @YalinLi0312,

This was just a toy example and I got it to work the way you described. 

In my productive setting I already receive Base64 encoded images from a previous API calling mine and would have liked it to pass it along to the Document Intelligence, hence why I posed the question. But decoding it and calling the Document Intelligence like that obviously works for me :)

Thanks a lot for your help! I'll close the issue!
